### PR TITLE
Update cache when awsconfig file changes

### LINKED ISSUE #2842: HostedUI uses cached configuration
**Describe the bug**
HostedUI configuration is loaded from SharedPreferences even when `awsconfiguration.json` is updated more recently.  This can manifest in a verification error on one of the initialization fields, even after changing the config file.

**To Reproduce**
1. Set up app with Hosted UI.
2. Set a breakpoint on `_showSignIn` in `AWSMobileClient.java`, run the app and note that it uses the proper configuration blob.
3. Update the `awsconfiguration.json` (and `amplifyconfiguration.json` -- you can run `grep` to make sure that there is no possible other source of backups), changing the `WebDomain`, for example.
4. Re-run the app and note that the blob has not updated.
5. Delete the virtual device and create a new one.
6. Re-run the app and note that the blob has updated.

**Which AWS service(s) are affected?**
After looking through `AWSMobileClient.java` (search for `mStore.get`), it looks like the issue is just with Hosted UI.

**Expected behavior**
Hosted UI should load the proper configuration from `awsconfiguration.json` and trust it over the SharedPreferences configuration.  I expect that this will just be a reordering of `getHostedUIJSON(awsConfig)`.

### PR DESCRIPTION
**Pull request by banji180**
*Issue #, if available:*
Fixes #2842

*Description of changes:*
Update cache when awsconfig file changes

Before:
Screen Shot 2022-07-15 at 5 21 48 AM

After:
Screen Shot 2022-07-15 at 9 37 27 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by tjleing — APPROVED** [review]
LGTM!  Feel free to put the screenshot on here or not

**Review by gpanshu — APPROVED** [review]

File: aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java [hunk 1]
```diff
@@ -705,19 +705,15 @@ JSONObject getHostedUIJSON() {
...
             JSONObject hostedUIJSON = null;
             try {
                 hostedUIJSON = new JSONObject(hostedUIString);
             } catch (Exception e) {
                 Log.w(TAG,
-                        "Failed to parse HostedUI settings from store. Defaulting to awsconfiguration.json", e);
+                        "Failed to parse HostedUI settings from store", e);
             }
-            if (hostedUIJSON == null && hostedUIJSONFromJSON != null) {
+            if (hostedUIJSONFromJSON != null && (hostedUIJSON == null || hostedUIJSON.toString() != hostedUIJSONFromJSON.toString())) {
```
Comment (div5yesh) [inline aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java:716]: What happens if the 2 config objects are equal?

[the remark this answers is not in the document]
Comment (div5yesh) [reply aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java:716]: +1

[the remark this answers is not in the document]
Comment (banji180) [reply aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java:716]: I tried writing test case but AWSMobileClientTest seems not to be set up properly. I have done the end-to-end testing as shown in screenshot attached and have also documented the change [here](https://quip-amazon.com/QFaUAfly1X76/Update-Config-Cache-when-Auth-file-changes).

**Review by sktimalsina — APPROVED** [review]

File: aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java (hunk 1 above)
Comment (banji180) [reply aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java:719]: Yes, the one in the cache is returned in that case.

**Review by div5yesh — APPROVED** [review]

**Pull request merged**